app/features/quizzify/tools.py

```diff
@@ -86,29 +94,266 @@ def load(self) -> List[Document]:
 
         return documents
 
-class BytesFilePDFLoader:
-    def __init__(self, files: List[Tuple[BytesIO, str]]):
+class BytesFileLoader:
```

Comment:
Recommend to use `BytesFileLoader` as the base class for a `BytesIO` file to process and return `langchain docs` then make subclasses which implement the method `load` to all of the subclasses and override the base method

Use DRY principle to make code more readable and less repeated

Reply:
I implemented new subclasses that inherit from the `BytesFileLoader` and have their own `load` method. I tried to combine similar document types together so that the code is more organized. Let me know if more organization is needed.